src/sagemaker/jumpstart/validators.py

```diff
@@ -25,44 +25,96 @@ def _validate_hyperparameter(
     ]
     if len(hyperparameter_spec) == 0:
         raise JumpStartHyperparametersError(
-            f"Unable to perform validation -- cannot find hyperparameter '{hyperparameter_name}' in model specs."
+            f"Unable to perform validation -- cannot find hyperparameter '{hyperparameter_name}' "
+            "in model specs."
+        )
+
+    if len(hyperparameter_spec) > 1:
+        raise JumpStartHyperparametersError(
+            f"Unable to perform validation -- found multiple hyperparameter "
+            f"'{hyperparameter_name}' in model specs."
         )
+
     hyperparameter_spec = hyperparameter_spec[0]
 
-    if hyperparameter_spec.type == VariableTypes.TEXT.value:
-        if type(hyperparameter_value) != str:
+    if hyperparameter_spec.type == VariableTypes.BOOL.value:
+        if isinstance(hyperparameter_value, bool):
+            return
+        if not isinstance(hyperparameter_value, str):
+            raise JumpStartHyperparametersError(
+                f"Expecting boolean valued hyperparameter, but got '{str(hyperparameter_value)}'."
+            )
+        if str(hyperparameter_value).lower() not in ["true", "false"]:
             raise JumpStartHyperparametersError(
-                f"Expecting text valued hyperparameter to have string type."
+                f"Expecting boolean valued hyperparameter, but got '{str(hyperparameter_value)}'."
+            )
+    elif hyperparameter_spec.type == VariableTypes.TEXT.value:
+        if not isinstance(hyperparameter_value, str):
+            raise JumpStartHyperparametersError(
+                "Expecting text valued hyperparameter to have string type."
             )
 
-        if getattr(hyperparameter_spec, "options", None):
+        if hasattr(hyperparameter_spec, "options"):
             if hyperparameter_value not in hyperparameter_spec.options:
                 raise JumpStartHyperparametersError(
-                    f"Hyperparameter '{hyperparameter_name}' must have one of the following values: "
-                    ", ".join(hyperparameter_spec.options)
+                    f"Hyperparameter '{hyperparameter_name}' must have one of the following "
+                    f"values: {', '.join(hyperparameter_spec.options)}"
+                )
+
+        if hasattr(hyperparameter_spec, "min"):
+            if len(hyperparameter_value) < hyperparameter_spec.min:
+                raise JumpStartHyperparametersError(
+                    f"Hyperparameter '{hyperparameter_name}' must have length no less than "
+                    f"{hyperparameter_spec.min}"
+                )
+
+        if hasattr(hyperparameter_spec, "exclusive_min"):
+            if len(hyperparameter_value) <= hyperparameter_spec.exclusive_min:
+                raise JumpStartHyperparametersError(
+                    f"Hyperparameter '{hyperparameter_name}' must have length greater than "
+                    f"{hyperparameter_spec.exclusive_min}"
+                )
```

Comment:
non-blocking: there is no real use-case for exclusive min/max with discreet values like integer (in this case text size), it's only useful for continuous values like float or double. That said, it doesn't really hurt to keep it either, it just makes the code a bit longer.